registry.go

```diff
@@ -0,0 +1,168 @@
+package lxd
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type imageServer struct {
+	Name          string              `json:"name"`
+	Desc          string              `json:"description"`
+	DescFr        string              `json:"description.fr"`
+	Url           string              `json:"url"`
+	Arguments     []map[string]string `json:"arguments"`
+	TrustedKeys   []string            `json:"trusted_keys"`
+	TrustedCerts  []string            `json:"trusted_certs"`
+	MinClientVer  int                 `json:"min_client_ver"`
+	MaxClientVer  int                 `json:"max_client_ver"`
+	imagePathList map[string]string
+}
+
+type registryServer struct {
+	Version         int           `json:"version"`
+	GenAt           uint64        `json:"generated_at"`
+	ImageServerList []imageServer `json:"servers"`
+}
+
+type RegistryManager struct {
+	serverCount int
+	client      *http.Client
+	registry    registryServer
+}
+
+func (rm *RegistryManager) InitRegistryManager() {
+	proxyUrl, err := url.Parse("http://172.26.0.75:8080")
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		Proxy:           http.ProxyURL(proxyUrl),
+	}
+	rm.client = &http.Client{Transport: tr}
+
+}
+
+func (rm *RegistryManager) FetchImageServerData() error {
+
+	resp, err := rm.client.Get("https://registry.linuxcontainers.org/1.0/index.json")
```

Comment:
Actually, I wanted to use NewClient() and ListContainer() functionality of client.go, but due to finger auth and ListContainer() URL expansion, I was not able to use it. 
So I wrote sample code to verify the functionality.